packages/terra-infinite-list/src/InfiniteList.jsx

```diff
@@ -0,0 +1,420 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import ResizeObserver from 'resize-observer-polyfill';
+import List from 'terra-list';
+import SelectableUtils from 'terra-list/lib/SelectableUtils';
+import InfiniteUtils from './_InfiniteUtils';
+import styles from './InfiniteList.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The child terra-list ListItems to be placed within the infinite list.
+   * ListItem is exported for consumption in dot notation from InfiniteList. (e.g. InfiniteList.Item)
+   */
+  children: PropTypes.node,
+  /**
+   * Whether or not unselected items should be disabled.
+   * Helpful for enabling max row selection.
+   */
+  disableUnselectedItems: PropTypes.bool,
+  /**
+   * Whether or not the child list items has a disclosure indicator presented.
+   * The behavior is intended to be used with a single selection style list, as multi selection style list should not perform disclosures.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * Properties related to the list styling.
+   */
+  infiniteProps: PropTypes.shape({
```

Comment:
IMO we should flatten these props

Reply:
Agreed, looking at the props table doc, I think this will be easier to read about if we flatten these props.